### ISSUE TITLE
Add a loading wheel

### ISSUE THREAD

**Issue by neha-deshpande001** [PollBuddy/PollBuddy#186]
**Please describe what has to be done**
When a page's data has not come in, display a loading wheel or some type of graphic.

```
    render() {
        if(this.state === null){
            //show nothing (or loading wheel) if the data has not come in yet
            return null;//loading todo ui
        }else{
            return (
                ...
            )
        }
```

**Comment by fanga2** [comment]
I'm finished with adding the wheel but I can't test it due to lack of an actual class editor. Should I make a pull request anyways or just ignore this until we implement the class editor?

**Comment by hbh7** [comment]
Can you simulate it showing up?

**Comment by fanga2** [comment]
Simulate it where exactly? Somewhere in the sitemap or somewhere else?

**Comment by hbh7** [comment]
Like on the classEditor, can you just add a variable that basically just controls whether or not the loader is displayed (for now, indefinitely) or not? Or am I misunderstanding? 

We won't be able to intentionally trigger/remove it for a while but that'd at least show it works/displays right

**Comment by fanga2** [comment]
I should be able to do that but it was my understanding that as of current there was no page that actually uses classEditor so we can't see if it displays right.

**Comment by hbh7** [comment]
It looks like classcreation implements (or at least looks like it implements) classeditor. Try experimenting with `<url>/classcreation`

**Comment by fanga2** [comment]
Ok in that case I can test it

**Comment by fanga2** [comment]
Alright, the loading wheel is tested and it can display (turn the loadingon variable to true in order to display it) should I make a pull request or wait until we can test to trigger/remove it?

**Comment by hbh7** [comment]
(records keeping)
We talked about and demo'd this during the 07/21/2020 meeting and have decided it would be best to turn this into a component, and then it'll be ready to pull request.

**Comment by fanga2** [comment]
I accidentally merged master instead of frontend into my branch. Should I try and reverse the merge (I figured out how to do this), make a pull request to master instead, or just make a pull request to frontend with all the changes of master added in?

**Comment by hbh7** [comment]
Generally I find it's safest to avoid merging master/frontend/backend into feature branches, but I think it usually works itself out when they get merged back. I'd say leave it as it is and we'll worry about it at PR/merge time. I'll make a note to bring frontend/backend/master inline first though just in case, then all _should_ be fine. 

TL;DR: Leave it as is, PR to frontend, and it should probably all just magically work out :)

**Comment by hbh7** [comment]
Complete in #233
